wasm/jit/engine.go

```diff
@@ -446,3 +447,230 @@ func (e *engine) builtinFunctionMemoryGrow(f *compiledWasmFunction) {
 func (e *engine) builtinFunctionMemorySize(f *compiledWasmFunction) {
 	e.push(uint64(len(f.memory.Buffer)) / wasm.PageSize)
 }
+
+func (e *engine) compileWasmFunction(f *wasm.FunctionInstance) (*compiledWasmFunction, error) {
+	ir, err := wazeroir.Compile(f)
+	if err != nil {
+		return nil, fmt.Errorf("failed to lower to wazeroir: %w", err)
+	}
+
+	compiler, err := newCompiler(e, f, ir)
+	if err != nil {
+		return nil, fmt.Errorf("failed to initialize assembly builder: %w", err)
+	}
+
+	compiler.emitPreamble()
+
+	for _, op := range ir.Operations {
+		switch o := op.(type) {
+		case *wazeroir.OperationUnreachable:
+			compiler.compileUnreachable()
+		case *wazeroir.OperationLabel:
+			if err := compiler.compileLabel(o); err != nil {
```

Comment:
what if you made err a variable then if set wrapped it based on the string of op?

Reply:
consolidated the error handling! thanks!